agent/consul/state/catalog.go

```diff
@@ -1938,6 +1938,41 @@ func (s *Store) CheckConnectServiceNodes(ws memdb.WatchSet, serviceName string,
 	return s.checkServiceNodes(ws, serviceName, true, entMeta)
 }
 
+// CheckGatewayServiceNodes is used to query all nodes and checks for services linked to a gateway
+func (s *Store) CheckGatewayServiceNodes(ws memdb.WatchSet, gateway string, entMeta *structs.EnterpriseMeta) (uint64, structs.CheckServiceNodes, error) {
```

Comment:
Should these be called CheckGatewayServicesNodes instead? (plural services)

Might make more sense since a gateway can be linked to multiple service names.

Reply:
CheckServiceNodes still have multiple service instances and it is more that the whole output is a`[]CheckServiceNode` wrapped in another struct to hold some index info.

So I think the name is fine as is.